src/NUnitFramework/framework/Constraints/Comparers/StreamsComparer.cs

```diff
@@ -39,8 +39,13 @@ public static EqualMethodResult Equal(object x, object y, ref Tolerance toleranc
                     return EqualMethodResult.ComparedEqual;
             }
 
+#if NETFRAMEWORK
             byte[] bufferExpected = new byte[BUFFER_SIZE];
             byte[] bufferActual = new byte[BUFFER_SIZE];
+#else
+            Span<byte> bufferExpected = stackalloc byte[BUFFER_SIZE];
+            Span<byte> bufferActual = stackalloc byte[BUFFER_SIZE];
```

Comment:
The BUFFER_SIZE is 4096.  That is a bit more than what is recommended for stackalloc, which says up to 1KB.   Should this be a concern here?

Reply:
I agree, even 1KB is on the higher side as I've seen 256 bytes often as a limit to stackalloc.

One other option here could be `ArrayPool`.

Reply:
I'll look into `ArrayPool` - haven't used it before so will investigate and come back later today.